subxt/src/client.rs

```diff
@@ -194,6 +196,22 @@ impl<T: Config> Client<T> {
     pub fn to_runtime_api<R: From<Self>>(self) -> R {
         self.into()
     }
+
+    /// Obtain the pallet unique identifier.
+    pub fn pallet_uid(&self, name: &'static str) -> Result<[u8; 32], MetadataError> {
+        let metadata = self.metadata.runtime_metadata();
+        // Note: fetch name from codegen to avoid `to_snake_case()`.
+        let pallet = match metadata
+            .pallets
+            .iter()
+            .find(|pallet| pallet.name.to_snake_case() == name)
+        {
+            Some(pallet) => pallet,
+            _ => return Err(MetadataError::PalletNotFound(name.to_string())),
+        };
+
+        Ok(self.metadata.metadata_hashable().get_pallet_uid(pallet))
+    }
```

Comment:
This feels like it would be better existing on `Metadata` rather than here on `Client`, since it doesn't need access to anything outside of `Metadata`

Reply:
(also, see other comment about `_hash` rather than `_uid`. Also, do we need to expose this? It may prove useful, but I'd probably keep it hidden away in the API unless there's an obvcious use for it)